### ISSUE TITLE
identity: deadlock with kvstore and identity-allocation-mode=crd

### ISSUE THREAD

**Issue by errordeveloper** [cilium/cilium#11820]
As discovered in https://github.com/cilium/cilium/issues/11690, there appears to be a deadlock when `identity-allocation-mode=crd` is used in combination with kvstore. Personally I don't know if this combination of options should be permitted, just reporting this as a follow-up, as right now it's possible for user to set it up this way and `cilium-agent` will lock-up (https://github.com/cilium/cilium/issues/11690#issuecomment-637044512).

**Comment by mrostecki** [comment]
I think that CRD and kvstore should be mutually exclusive. @aanm, do you agree?

**Comment by aanm** [comment]
@mrostecki using a kvstore with CRD can be allowed AFAIK. Also I think the agent lock up due https://github.com/cilium/cilium/issues/11836

@errordeveloper is there a way to reproduce this issue?

**Comment by errordeveloper** [comment]
> @errordeveloper is there a way to reproduce this issue?

You can revert part of 6fe8d092a6796d36230a0e3480cd2a8472e1a6f0, i.e.:
```
diff --git a/test/k8sT/DatapathConfiguration.go b/test/k8sT/DatapathConfiguration.go
index 7466a9dfe..2550be178 100644
--- a/test/k8sT/DatapathConfiguration.go
+++ b/test/k8sT/DatapathConfiguration.go
@@ -560,9 +560,8 @@ var _ = Describe("K8sDatapathConfig", func() {
 
                        etcdService := fmt.Sprintf("http://%s:%d", host, port)
                        opts := map[string]string{
-                               "global.etcd.enabled":           "true",
-                               "global.etcd.endpoints[0]":      etcdService,
-                               "global.identityAllocationMode": "kvstore",
+                               "global.etcd.enabled":      "true",
+                               "global.etcd.endpoints[0]": etcdService,
                        }
                        if helpers.ExistNodeWithoutCilium() {
                                opts["global.synchronizeK8sNodes"] = "false"
```
And then focus on this particular test. I recall reproducing it on first attempt in GKE.

For the record, this is what I did:
```
ginkgo -v --focus="K8sDatapathConfig Etcd" -noColor -- -cilium.provision=false -cilium.testScope=K8s -cilium.kubeconfig=$KUBECONFIG -cilium.image="$CILIUM_IMAGE" -cilium.operator-image="$CILIUM_OPERATOR_IMAGE" -cilium.passCLIEnvironment=true -cilium.skipLogs 
```
I've used latest images at the time, cannot say exactly what those were, but in this case it shouldn't matter.

**Comment by errordeveloper** [comment]
We should confirm if this was fixed by #11950 or not.
